include/alpaka/ctx/block/CtxBlockOacc.hpp

```diff
@@ -228,10 +228,17 @@ namespace alpaka
             //-----------------------------------------------------------------------------
             static auto declareVar(CtxBlockOacc<TDim, TIdx> const& smem) -> T&
             {
-                traits::SyncBlockThreads<CtxBlockOacc<TDim, TIdx>>::masterOpBlockThreads(smem, [&smem]() {
-                    smem.template alloc<T>();
-                });
-                return smem.template getLatestVar<T>();
+                auto* data = smem.template getVarPtr<T>(TuniqueId);
+
+                if(!data)
+                {
+                    traits::SyncBlockThreads<CtxBlockOacc<TDim, TIdx>>::masterOpBlockThreads(smem, [&data, &smem]() {
+                        smem.template alloc<T>(TuniqueId);
+                    });
+                    data = smem.template getLatestVarPtr<T>();
+                }
+                ALPAKA_ASSERT(data == nullptr);
+                return std::ref(*data);
```

Comment:
Why not `return *data`?

Reply:
It was in old implementations like this. https://github.com/alpaka-group/alpaka/pull/1258/files/55e205b395a5b780430780ee256e0a716e175feb#diff-33fbcb49de69fef7bf79c86c0106d8497046ffd67dfc8305c1912987a6a16585L81

I tried to understand why it was returning a `std::ref`. Since I was not sure I keep it. @BenjaminW3 could you explain why the old implementation returned a `std::ref`

Reply:
I think with the signature of the function returning `T&` explicitly, so as `auto ... -> T&`, and so it's better to remove `std::ref` from return of such funcitons. It indeed does nothing and just causes confusion. If there is a function what just has `auto` as return type and returns `std::ref` it would be another story.